Source/DafnyRuntime/DafnyRuntimePython/_dafny/__init__.py

```diff
@@ -116,11 +116,71 @@ def flatten(self):
             e = q.pop()
             if isinstance(e, list):
                 l += e
+            elif isinstance(e, _SeqSlice):
+                l += e
             elif isinstance(e, Concat):
                 q.append(e.r)
                 q.append(e.l)
         return l
 
+class _SeqSlice:
```

Comment:
It's a little strange, that we have `Concat` and `_SeqSlice`. Unifying the naming scheme seems appropriate.

Reply:
I'm fine either way

Reply:
Changed to Slice